beacon-chain/powchain/engine-api-client/v1/client.go

```diff
@@ -277,3 +280,15 @@ func handleRPCError(err error) error {
 		return err
 	}
 }
+
+var ErrHTTPTimeout = errors.New("timeout from http.Client")
+
+type HTTPTimeoutError interface {
+	Error() string
+	Timeout() bool
+}
+
+func iHTTPTimeoutError(e error) bool {
```

Comment:
nitpick, function name looks weird, could just call it `isTimeout` which would be pretty clear in context. but i don't mind if you merge it as-is.